### PR TITLE
Add emit delta i8

### PR DESCRIPTION
**Pull request by ameykasbe**

### CODE REVIEW

File: tsz-compress/tests/compress-v2.rs [hunk 1]
```diff
@@ -3,6 +3,8 @@ use tsz_compress::prelude::*;
 
 extern crate alloc;
 
+use rand::Rng;
```
Comment (trueb2) [inline tsz-compress/tests/compress-v2.rs]: Prefer imports close to the module of use unless the intent is to export. This import would belong in the tests mod.

File: tsz-compress/tests/compress-v2.rs [hunk 2]
```diff
@@ -126,6 +216,189 @@ mod tests {
         assert_eq!(decompressor.col_a(), vec![row.a; N]);
     }
 
+    #[test]
+    fn test_macro_compress_test_macro_compress_random_i32() {
```
Comment (trueb2) [inline tsz-compress/tests/compress-v2.rs]: test_macro_compress_test_macro_compress_random_i32

These test names are redundant

File: tsz-compress/tests/compress-v2.rs [hunk 3]
```diff
@@ -126,6 +216,189 @@ mod tests {
...
+            // Initialize the input vector
+            let mut values: Vec<i32> = vec![];
+
+            // Number of samples in the input vector
+            let end_range = rng.gen_range(100..10000);
+
+            // Generate input vector randomly in i32 range
+            for _ in 0..end_range {
+                values.push(rng.gen_range(i32::MIN..i32::MAX));
+            }
```
Comment (trueb2) [inline tsz-compress/tests/compress-v2.rs]: This is test code, but we can be more concise and efficient with iterators. 

Using idiomatic code will result in fewer lines, fewer allocations, and faster tests.

```rust
    // Number of samples in the input vector
    let end_range = rng.gen_range(100..10000);

    // Generate input vector randomly in i32 range
    let values: Vec<i32> = (0..end_range)
        .map(|_| rng.gen_range(i32::MIN..i32::MAX))
        .collect();
```

Alternatively, the `rand` crate has features for this use case, which are likely to be best.
```rust
   // Number of samples in the input vector
    let end_range = rng.gen_range(100..10000);

    // Create a vector with the specified number of elements
    let mut values = vec![0i32; end_range];

    // Fill the vector with random i32 values
    rng.fill(values.as_mut_slice());
```

File: tsz-compress/tests/compress-v2.rs [hunk 4]
```diff
@@ -126,6 +216,189 @@ mod tests {
...
+        // Decompress the bit buffer
+        decompressor.decompress(&bytes).unwrap();
+
+        // Assert that the decompressed data matches the original
+        assert_eq!(decompressor.col_a(), a_col_vec);
+        assert_eq!(decompressor.col_b(), b_col_vec);
+        assert_eq!(decompressor.col_c(), c_col_vec);
+    }
+
+    // #[test]
```
Comment (trueb2) [inline tsz-compress/tests/compress-v2.rs]: Avoid adding or leaving dead code, especially when breaking up changes into smaller PRs.

**Review by trueb2 — CHANGES_REQUESTED** [review]

File: tsz-compress/tests/compress-v2.rs (hunk 1 above)
Comment (ameykasbe) [reply tsz-compress/tests/compress-v2.rs]: Added changes in commit: 
Refactor: Enhance Code Idiomaticity and Readability ([42dcf40](https://github.com/qsib-cbie/tsz/pull/25/commits/42dcf407a455aab897eb5925d96ab7a3c0547021))

File: tsz-compress/tests/compress-v2.rs (hunk 2 above)
Comment (ameykasbe) [reply tsz-compress/tests/compress-v2.rs]: Added changes in commit: 
Refactor: Enhance Code Idiomaticity and Readability ([42dcf40](https://github.com/qsib-cbie/tsz/pull/25/commits/42dcf407a455aab897eb5925d96ab7a3c0547021))

File: tsz-compress/tests/compress-v2.rs (hunk 3 above)
Comment (ameykasbe) [reply tsz-compress/tests/compress-v2.rs]: Added changes in commit: 
Refactor: Enhance Code Idiomaticity and Readability ([42dcf40](https://github.com/qsib-cbie/tsz/pull/25/commits/42dcf407a455aab897eb5925d96ab7a3c0547021))

File: tsz-compress/tests/compress-v2.rs (hunk 4 above)
Comment (ameykasbe) [reply tsz-compress/tests/compress-v2.rs]: Added changes in commit: 
Refactor: Enhance Code Idiomaticity and Readability ([42dcf40](https://github.com/qsib-cbie/tsz/pull/25/commits/42dcf407a455aab897eb5925d96ab7a3c0547021))

**Review by trueb2 — APPROVED** [review]

**Pull request merged**